examples/chat.rs

```diff
@@ -124,26 +126,19 @@ fn run(socket: UdpSocket, rx: Receiver<Rtc>) -> Result<(), RtcError> {
             clients.push(client);
         }
 
-        // Poll all clients, and get propagated events as a result.
-        let to_propagate: Vec<_> = clients.iter_mut().map(|c| c.poll_output(&socket)).collect();
-        let timeouts: Vec<_> = to_propagate.iter().filter_map(|p| p.as_timeout()).collect();
-
-        // We keep propagating client events until all clients respond with a timeout.
-        if to_propagate.len() > timeouts.len() {
-            propagate(&mut clients, to_propagate);
-            // Start over to propagate more client data until all are timeouts.
-            continue;
+        // Poll clients until they return timeout
+        let mut timeout = Instant::now() + Duration::from_millis(100);
+        for client in clients.iter_mut() {
+            let t = poll_until_timeout(client, &mut propagation, &socket);
+            timeout = timeout.min(t);
```

Comment:
Propagation must happen before `poll_until_timeout` or we might get starvation.

Reply:
Can you explain this ? We discover stuff to propagate while polling

Reply:
If polling creates more Propagated than we manage to send, we are starving the sending.

The sending side discards stuff, so we need to ensure we are keeping the propagation clear.

Reply:
Yes. I expect polling to create more propagated, and propagation to sometimes result in more stuff to "poll and propagate" again. 
In this phase we do not drive time forward and we do not check network input. 
Everything must settle (and it does)